### ISSUE TITLE
 Invalid argument: From /job:tpu_worker/replica:0/task:0:

### ISSUE THREAD

**Issue by joshclancy** [pytorch/xla#2109]
## ❓ Questions and Help
Hello!

I have run into a problem I do not fully understand. I have been successfully training on TPUs with xla (awesome!) but after adding some logging capabilities (of all things) my code breaks. I am not sure what's going on. Once it breaks it seems that I am unable to run anything without restarting the kernal.

The error write out makes it seem like it may be a wandb problem (weights and biases - my fav logging software) but then again the final part seems distinctly TPU related. Any ideas or pointers are welcome, I'm outside of my domain knowledge on this one.  

NAME    ZONE           ACCELERATOR_TYPE  NETWORK_ENDPOINTS  NETWORK  RANGE            STATUS
node-1  us-central1-f  v2-8              10.16.158.82:8470  default  10.16.158.80/29  READY
Error in callback <function _init_jupyter.<locals>.cleanup at 0x7f4f84efb6a8> (for post_run_cell):

---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
/anaconda3/envs/torch-xla-nightly/lib/python3.6/site-packages/backcall/backcall.py in adapted(*args, **kwargs)
    102                 kwargs.pop(name)
    103 #            print(args, kwargs, unmatched_pos, cut_positional, unmatched_kw)
--> 104             return callback(*args, **kwargs)
    105 
    106         return adapted

/anaconda3/envs/torch-xla-nightly/lib/python3.6/site-packages/wandb/__init__.py in cleanup()
    504         # shutdown async logger because _user_process_finished isn't called in jupyter
    505         shutdown_async_log_thread()
--> 506         run._stop_jupyter_agent()
    507     if hasattr(ipython.events, "_orig_post_run"):
    508         ipython.events.unregister("post_run_cell", ipython.events._orig_post_run)

/anaconda3/envs/torch-xla-nightly/lib/python3.6/site-packages/wandb/wandb_run.py in _stop_jupyter_agent(self)
    198 
    199     def _stop_jupyter_agent(self):
--> 200         self._jupyter_agent.stop()
    201 
    202     def send_message(self, options):

/anaconda3/envs/torch-xla-nightly/lib/python3.6/site-packages/wandb/jupyter.py in stop(self)
    139             self.save_history()
    140             self.rm.unmirror_stdout_stderr()
--> 141             wandb.run.close_files()
    142             self.rm.shutdown()
    143             self.paused = True

/anaconda3/envs/torch-xla-nightly/lib/python3.6/site-packages/wandb/wandb_run.py in close_files(self)
    759             self._events = None
    760         if self._history is not None:
--> 761             self._history.close()
    762             self._history = None
    763 

/anaconda3/envs/torch-xla-nightly/lib/python3.6/site-packages/wandb/history.py in close(self)
    258 
    259     def close(self):
--> 260         self._write()
    261         self._lock.acquire()
    262         try:

/anaconda3/envs/torch-xla-nightly/lib/python3.6/site-packages/wandb/history.py in _write(self)
    243                     self.row["_stream"] = self.stream_name
    244                 self._transform()
--> 245                 self._file.write(util.json_dumps_safer_history(self.row))
    246                 self._file.write('\n')
    247                 self._file.flush()

/anaconda3/envs/torch-xla-nightly/lib/python3.6/site-packages/wandb/util.py in json_dumps_safer_history(obj, **kwargs)
    486 def json_dumps_safer_history(obj, **kwargs):
    487     """Convert obj to json, with some extra encodable types, including histograms"""
--> 488     return json.dumps(obj, cls=WandBHistoryJSONEncoder, **kwargs)
    489 
    490 def make_json_if_not_number(v):

/anaconda3/envs/torch-xla-nightly/lib/python3.6/json/__init__.py in dumps(obj, skipkeys, ensure_ascii, check_circular, allow_nan, cls, indent, separators, default, sort_keys, **kw)
    236         check_circular=check_circular, allow_nan=allow_nan, indent=indent,
    237         separators=separators, default=default, sort_keys=sort_keys,
--> 238         **kw).encode(obj)
    239 
    240 

/anaconda3/envs/torch-xla-nightly/lib/python3.6/json/encoder.py in encode(self, o)
    197         # exceptions aren't as detailed.  The list call should be roughly
    198         # equivalent to the PySequence_Fast that ''.join() would do.
--> 199         chunks = self.iterencode(o, _one_shot=True)
    200         if not isinstance(chunks, (list, tuple)):
    201             chunks = list(chunks)

/anaconda3/envs/torch-xla-nightly/lib/python3.6/json/encoder.py in iterencode(self, o, _one_shot)
    255                 self.key_separator, self.item_separator, self.sort_keys,
    256                 self.skipkeys, _one_shot)
--> 257         return _iterencode(o, 0)
    258 
    259 def _make_iterencode(markers, _default, _encoder, _indent, _floatstr,

/anaconda3/envs/torch-xla-nightly/lib/python3.6/site-packages/wandb/util.py in default(self, obj)
    456 
    457     def default(self, obj):
--> 458         obj, converted = json_friendly(obj)
    459         obj, compressed = maybe_compress_history(obj)
    460         if converted:

/anaconda3/envs/torch-xla-nightly/lib/python3.6/site-packages/wandb/util.py in json_friendly(obj)
    316             obj = obj.numpy()
    317         else:
--> 318             return obj.item(), True
    319 
    320     if is_numpy_array(obj):

RuntimeError: Invalid argument: From /job:tpu_worker/replica:0/task:0:
Run-time shape mismatch for XRTExecute argument[229] (4368608986020068). Expected element_type: F32
dimensions: 1
layout {
  minor_to_major: 0
  format: DENSE
}
is_dynamic_dimension: false
; got element_type: F32
dimensions: 4
layout {
  minor_to_major: 0
  format: DENSE
}
is_dynamic_dimension: false

	 [[{{node XRTExecute}}]]

**Comment by taylanbil** [comment]
Could you please provide a small colab repro for this? It's not easy to debug from the traceback only.

**Comment by joshclancy** [comment]
sure give me a bit, I'll try and recreate the error in collab.

**Comment by joshclancy** [comment]
hmmm well I was unable to replicate on colab. 

I did however the problem to this line:
recon = outputDict['recon_x_list'][n][0].cpu().detach().numpy()
where I have a tensor within a list within a dictionary. Weirdly this line is within a loop and the error triggers on the second pass of the loop. 

Either way I found a hacky way around the problem by getting rid of the loop but I still don't fully understand what was going on.  

Thanks for the help.
